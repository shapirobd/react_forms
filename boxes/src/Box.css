.Box {
	margin-right: 5px;
	position: relative;
	display: inline-block;
	vertical-align: middle;
}

.Box-container {
	margin: 5px auto;
	height: fit-content;
}

.Box-btn {
	position: relative;
	display: inline-block;
	vertical-align: middle;
}
