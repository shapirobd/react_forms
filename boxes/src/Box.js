import React from "react";
import "./Box.css";

const Box = ({ color, width, height, handleDelete }) => {
	return (
		<div className="Box-container">
			<div
				className="Box"
				style={{ backgroundColor: color, width: width, height: height }}
			></div>
			<button className="Box-btn" onClick={handleDelete}>
				X
			</button>
		</div>
	);
};

export default Box;
