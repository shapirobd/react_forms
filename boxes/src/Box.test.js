import { render, screen } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", () => {
	render(<Box />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<Box />);
	expect(asFragment()).toMatchSnapshot();
});

it("has correct color, height and width", () => {
	const { getByRole } = render(<Box color="blue" width="45px" height="30px" />);
	expect(getByRole("box")).toHaveStyle("background-color : blue");
	expect(getByRole("box")).toHaveStyle("width : 45px");
	expect(getByRole("box")).toHaveStyle("height : 30px");
});
