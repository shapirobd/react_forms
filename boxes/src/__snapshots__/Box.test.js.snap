// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has correct color, height and width 1`] = `
<DocumentFragment>
  <div
    class="Box-container"
  >
    <div
      class="Box"
      role="box"
    />
    <button
      class="Box-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-container"
  >
    <div
      class="Box"
      role="box"
    />
    <button
      class="Box-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
